pkg/model/domodel/droplets.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package domodel
+
+import (
+	"strings"
+
+	"k8s.io/kops/pkg/model"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/dotasks"
+)
+
+// DropletBuilder configures droplets for the cluster
+type DropletBuilder struct {
+	*DOModelContext
+
+	BootstrapScript *model.BootstrapScript
+	Lifecycle       *fi.Lifecycle
+}
+
+var _ fi.ModelBuilder = &DropletBuilder{}
+
+func (d *DropletBuilder) Build(c *fi.ModelBuilderContext) error {
+	sshKeyName, err := d.SSHKeyName()
+	if err != nil {
+		return err
+	}
+
+	splitSSHKeyName := strings.Split(sshKeyName, "-")
+	sshKeyFingerPrint := splitSSHKeyName[len(splitSSHKeyName)-1]
+
+	// replace "." with "-" since DO API does not accept "."
```

Comment:
GCE has a similar issue, and has `SafeClusterName` - you might want to create a similar function for DO, depending on how often you end up using it!

Reply:
ahh, that's useful noted!